actionpack/lib/action_controller/metal/redirecting.rb

```diff
@@ -82,8 +82,8 @@ def redirect_to(options = {}, response_status = {})
     #
     # All options that can be passed to <tt>redirect_to</tt> are accepted as
     # options and the behavior is identical.
-    def redirect_back(fallback_location:, **args)
-      if referer = request.headers["Referer"]
+    def redirect_back(fallback_location:, allow_other_host: true, **args)
+      if (referer = request.headers["Referer"]) && (allow_other_host || URI(referer).host == request.host)
```

Comment:
`URI(referer)` can raise exceptions. Should we make sure that returns `false`?

Reply:
Indeed, it raises an error on non-string attribute.
So what if I use `URI(referer.to_s).host == request.host`?

Reply:
I think there are more cases where it could raise. I'd prefer to call `to_s` but also rescue the possible exceptions and return `false` in those cases.